build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmExtractorConsumer.java

```diff
@@ -61,12 +64,50 @@ public void consumerRun() {
     }
 
     /**
-     * Create 'Dependency' from name and version of 'npmPackageInfo'. Try to retrieve sha1 and md5 from Artifactory.
+     * If package contained in the dependencies map, add the current scope for the dependency.
+     * Otherwise, retrieve sha1 and md5 from Artifactory and add the dependency to the dependencies map.
+     *
+     * @param npmPackageInfo - The npm package information.
+     * @return True if the package is legal. False in case of an error such as an absence in Artifactory's cache.
+     */
+    private boolean appendDependency(NpmPackageInfo npmPackageInfo) {
+        String id = npmPackageInfo.getName() + ":" + npmPackageInfo.getVersion();
+        if (!dependencies.containsKey(id)) {
+            Dependency dependency = createDependency(npmPackageInfo, id);
+            if (dependency == null) {
+                return false;
+            }
+            dependencies.put(id, dependency);
+        } else {
+            dependencies.get(id).getScopes().add(npmPackageInfo.getScope());
+        }
+        return true;
+    }
+
+    /**
+     * Create a Dependency for the provided NpmPackageInfo.
+     * If the dependency exists in the previous build's dependencies - take the required info from it.
+     * Otherwise - fetch the information from Artifactory.
      *
      * @param npmPackageInfo - The npm package information.
+     * @param id             - The id of the dependency to create.
      * @return Dependency populated with {name, scope, version, sha1 and md5} or null in case of {error or absence in Artifactory's case}.
      */
-    private Dependency createDependency(NpmPackageInfo npmPackageInfo) {
+    private Dependency createDependency(NpmPackageInfo npmPackageInfo, String id) {
+        if (previousBuildDependencies.containsKey(id)) {
+            return createDependencyFromPreviousBuild(npmPackageInfo, id);
+        }
+        return createDependencyFromAqlResult(npmPackageInfo, id);
+    }
+
+    /**
+     * Create 'Dependency' from name and version of 'npmPackageInfo'. Try to retrieve sha1 and md5 from Artifactory.
+     *
+     * @param npmPackageInfo - The npm package information.
+     * @param id             - The id of the dependency to create.
+     * @return Dependency or null in case of {error or absence in Artifactory's case}.
```

Comment:
Dependency or null in case of {error or absence in Artifactory's case}. -->
Dependency or null, in case of an exception. or in case the dependency does not exist Artifactory.